hypertrace-graphql-entity-schema/src/main/java/org/hypertrace/graphql/entity/dao/GatewayServiceEntityDao.java

```diff
@@ -52,31 +62,50 @@
                 grpcChannelRegistry.forAddress(
                     serviceConfig.getGatewayServiceHost(), serviceConfig.getGatewayServicePort()))
             .withCallCredentials(credentials);
+
+    this.labelConfigServiceStub =
+        LabelsConfigServiceGrpc.newFutureStub(
+                grpcChannelRegistry.forAddress(
+                    serviceConfig.getConfigServiceHost(), serviceConfig.getConfigServicePort()))
+            .withCallCredentials(credentials);
   }
 
   @Override
   public Single<EntityResultSet> getEntities(EntityRequest request) {
-    return this.requestBuilder
-        .buildRequest(request)
-        .subscribeOn(this.boundedIoScheduler)
-        .flatMap(
-            serverRequest ->
-                this.makeRequest(request.resultSetRequest().context(), serverRequest)
-                    .flatMap(
-                        serverResponse ->
-                            baselineDao
-                                .getBaselines(
-                                    request.resultSetRequest().context(),
-                                    serverRequest,
-                                    serverResponse,
-                                    request)
-                                .flatMap(
-                                    baselineResponse ->
-                                        this.entityConverter.convert(
-                                            request, serverResponse, baselineResponse))));
+    return Single.zip(
+            this.requestBuilder
+                .buildRequest(request)
+                .subscribeOn(this.boundedIoScheduler)
+                .flatMap(
+                    serverRequest ->
+                        this.makeEntityRequest(request.resultSetRequest().context(), serverRequest)
+                            .map(serverResponse -> Map.entry(serverRequest, serverResponse))),
+            this.makeLabelRequest(request.resultSetRequest().context(), request)
+                .subscribeOn(this.boundedIoScheduler),
+            (serviceRequestResponseEntry, labelsResponse) -> {
+              EntitiesRequest entitiesRequest = serviceRequestResponseEntry.getKey();
+              EntitiesResponse entitiesResponse = serviceRequestResponseEntry.getValue();
+              Map<String, String> labelsMap =
+                  labelsResponse
+                      .map(GetLabelsResponse::getLabelsList)
+                      .orElse(Collections.emptyList())
+                      .stream()
+                      .collect(Collectors.toUnmodifiableMap(Label::getId, Label::getKey));
+              return baselineDao
```

Comment:
Can we please extract this out into a private function for better readability of the function?

Reply:
done